cmd/proxy-test-server/proxy-test-server.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// The proxy-test-server command is a simple HTTP proxy server for testing
+// Tailscale's client proxy functionality.
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"flag"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"golang.org/x/crypto/acme/autocert"
+	"tailscale.com/tempfork/acme"
+)
+
+var (
+	listen        = flag.String("listen", ":8080", "Address to listen on for HTTPS proxy requests")
+	hostname      = flag.String("hostname", "localhost", "Hostname for the proxy server")
+	tailscaleOnly = flag.Bool("tailscale-only", true, "Restrict proxy to Tailscale targets only")
+	allowHosts    = flag.String("allow-hosts", "", "Comma-separated list of allowed target hosts to additionally allow")
+)
+
+func main() {
+	flag.Parse()
+
+	am := &autocert.Manager{
+		HostPolicy: autocert.HostWhitelist(*hostname),
+		Prompt:     autocert.AcceptTOS,
+		Cache:      autocert.DirCache(os.ExpandEnv("$HOME/.cache/autocert/proxy-test-server")),
+	}
+	var allowTarget func(hostPort string) bool
+	if *tailscaleOnly {
+		allowTarget = func(hostPort string) bool {
+			host, port, err := net.SplitHostPort(hostPort)
+			if err != nil {
+				log.Printf("invalid target %q: %v", hostPort, err)
+				return false
```

Comment:
optional: maybe return an `error` instead of a `bool` here and let the caller log?